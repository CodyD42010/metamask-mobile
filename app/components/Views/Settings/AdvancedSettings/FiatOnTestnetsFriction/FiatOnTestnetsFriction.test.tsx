import React from 'react';
import { render } from '@testing-library/react-native';
import FiatOnTestnetsFriction from './FiatOnTestnetsFriction';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

const store = configureMockStore()({});
jest.mock('@react-navigation/native');
jest.mock('react-native-safe-area-context', () => ({
  useSafeAreaInsets: jest.fn().mockImplementation(() => ({})),
  useSafeAreaFrame: jest.fn().mockImplementation(() => ({})),
}));

describe('Show fiat on testnets friction bottom sheet', () => {
  it('should render', () => {
    const wrapper = render(
      <Provider store={store}>
        <FiatOnTestnetsFriction />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
