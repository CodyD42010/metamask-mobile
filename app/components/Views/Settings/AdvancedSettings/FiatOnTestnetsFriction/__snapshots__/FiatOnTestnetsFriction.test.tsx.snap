// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show fiat on testnets friction bottom sheet should render 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "bottom": 0,
        "justifyContent": "flex-end",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#00000099",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "opacity": 0,
        },
      ]
    }
  >
    <TouchableOpacity
      onPress={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </View>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      collapsable={false}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onLayout={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "borderColor": "#D6D9DC",
            "borderTopLeftRadius": 8,
            "borderTopRightRadius": 8,
            "borderWidth": 1,
            "maxHeight": NaN,
            "overflow": "hidden",
            "paddingBottom": undefined,
            "shadowColor": "#0000001A",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 40,
          },
          Object {
            "transform": Array [
              Object {
                "translateY": 1334,
              },
            ],
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "padding": 4,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#D6D9DC",
              "borderRadius": 2,
              "height": 4,
              "width": 40,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "padding": 16,
          }
        }
      >
        <SvgMock
          color="#FFC70A"
          height={32}
          name="Danger"
          style={
            Object {
              "height": 32,
              "width": 32,
            }
          }
          width={32}
        />
        <Text
          style={
            Array [
              Object {
                "color": "#24272A",
                "fontFamily": "EuclidCircularB-Regular",
                "fontSize": 30,
                "fontWeight": "400",
                "marginVertical": 2,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "fontFamily": "Euclid Circular B",
                "fontSize": 18,
                "fontWeight": "700",
                "letterSpacing": 0,
                "lineHeight": 24,
              },
            ]
          }
        >
          Be careful
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#24272A",
                "fontFamily": "EuclidCircularB-Regular",
                "fontSize": 30,
                "fontWeight": "400",
                "marginVertical": 2,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "marginTop": 16,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#24272A",
                  "fontFamily": "EuclidCircularB-Regular",
                  "fontSize": 30,
                  "fontWeight": "400",
                  "marginVertical": 2,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            If you've been asked to turn this feature on, you might be getting scammed. These tokens have no monetary value and are for testing purposes only. This feature helps developers make sure their apps work.
             
          </Text>
          <Text
            accessibilityRole="link"
            accessible={true}
            onPress={[Function]}
            style={
              Array [
                Object {
                  "color": "#24272A",
                  "fontFamily": "EuclidCircularB-Regular",
                  "fontSize": 30,
                  "fontWeight": "400",
                  "marginVertical": 2,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "color": "#24272A",
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "color": "#0376C9",
                },
                undefined,
                undefined,
                Object {
                  "marginVertical": 0,
                },
                undefined,
              ]
            }
          >
            Learn more.
          </Text>
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
              "gap": 16,
              "justifyContent": "space-evenly",
              "marginTop": 32,
            }
          }
        >
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={1}
            onPress={[Function]}
            onPressIn={[Function]}
            onPressOut={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "backgroundColor": "transparent",
                "borderColor": "#0376C9",
                "borderRadius": 24,
                "borderWidth": 1,
                "flex": 1,
                "flexDirection": "row",
                "height": 48,
                "justifyContent": "center",
                "paddingHorizontal": 16,
              }
            }
            testID="cancel-button"
            variant="Secondary"
          >
            <Text
              accessibilityRole="text"
              style={
                Object {
                  "color": "#0376C9",
                  "fontFamily": "Euclid Circular B",
                  "fontSize": 14,
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 22,
                }
              }
            >
              Cancel
            </Text>
          </TouchableOpacity>
          <TouchableOpacity
            accessibilityRole="button"
            accessible={true}
            activeOpacity={1}
            onPress={[Function]}
            onPressIn={[Function]}
            onPressOut={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "backgroundColor": "#0376C9",
                "borderRadius": 24,
                "flex": 1,
                "flexDirection": "row",
                "height": 48,
                "justifyContent": "center",
                "paddingHorizontal": 16,
              }
            }
            testID="continue-button"
            variant="Primary"
          >
            <Text
              accessibilityRole="text"
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "Euclid Circular B",
                  "fontSize": 14,
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 22,
                }
              }
            >
              Continue
            </Text>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  </View>
</View>
`;
